language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/signature/SignatureInfoModelBuilder.java

```diff
@@ -182,11 +181,11 @@ private void fillParamInfoModels() {
             return;
         }
 
-        List<Parameter> parameters = parameterSymbols
+        List<Parameter> parameters = new ArrayList<>(parameterSymbols
```

Comment:
If we need this to be mutable let's leave it as it was previously. But I think we can move this after `restParam` (since it don't depend on it) and make this immutable

Reply:
I made it immutable. Fixed in [2778660](https://github.com/ballerina-platform/ballerina-lang/pull/42904/commits/277866060d74b08115089625bb7d582c095697a1)